auth/authController.js

```diff
@@ -7,7 +7,7 @@ const secretWord = process.env.SECRET;
 
 const auth = (req, res, next) => {
   passport.authenticate("jwt", { session: false }, (err, user) => {
-    if (!user || err) {
+    if (!user || err || !user.token) {
```

Comment:
Ok wyobraź sobie teraz sytuację że zalogujesz się 2 razy wtedy stary token nadpisze się nowym w bazie danych. Przy autoryzacji starym tokenem wyjdzie że jakiś token jest w bazie więc przepuści to żądanie i uzna je za poprawne.
w req masz token i możesz go tu porównać z tokenem w zwracanym userze będzie to wyglądać mniejwięcej tak

const token = req.headers.authorization.split(' ')[1];
if (!user || err || user.token !== token) {
.....
}